src/main/java/ch/njol/skript/expressions/ExprNewBannerPattern.java

```diff
@@ -0,0 +1,83 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.block.banner.PatternType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Banner Pattern")
+@Description("Create a new banner pattern.")
+@Examples({
+	"set {_pattern} to a creeper banner pattern colored red",
+	"add {_pattern} to banner patterns of {_banneritem}",
+	"remove {_pattern} from banner patterns of {_banneritem}",
+	"set the 1st banner pattern of block at location(0,0,0) to {_pattern}",
+	"clear the 1st banner pattern of block at location(0,0,0)",
+	"",
+	"set {_pattern} to a red mojang banner pattern"
+})
+@Since("INSERT VERSION")
+public class ExprNewBannerPattern extends SimpleExpression<Pattern> {
+
+	static {
+		Skript.registerExpression(ExprNewBannerPattern.class, Pattern.class, ExpressionType.PATTERN_MATCHES_EVERYTHING,
+			"[a] %bannerpatterntype% colo[u]red %color%",
+			"[a] %*color% %bannerpatterntype%");
+	}
+
+	private Expression<PatternType> selectedPattern;
+	private Expression<Color> selectedColor;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 0) {
+			//noinspection unchecked
+			selectedPattern = (Expression<PatternType>) exprs[0];
+			//noinspection unchecked
+			selectedColor = (Expression<Color>) exprs[1];
+		} else {
+			//noinspection unchecked
+			selectedPattern = (Expression<PatternType>) exprs[1];
+			//noinspection unchecked
+			selectedColor = (Expression<Color>) exprs[0];
+		}
```

Comment:
```suggestion
		//noinspection unchecked
		selectedPattern = (Expression<PatternType>) exprs[matchedPattern == 0 ? 0 : 1];
		//noinspection unchecked
		selectedColor = (Expression<Color>) exprs[matchedPattern == 0 ? 1 : 0];
```

Reply:
Could I void this one? I personally dont like the look of using a ternary inside the array like this.